### ISSUE TITLE
数值不稳定问题

### ISSUE THREAD

**Issue by yangwei-nlp** [yatengLG/Focal-Loss-Pytorch#7]
由于数值溢出问题，当preds中错误类别的数值绝对值特别大的时候，loss容易计算得到`inf`。

建议不要采用`softmax+log`，采用`log_softmax+exp`会更好。

https://github.com/yatengLG/Focal-Loss-Pytorch/blob/1f3da28a716121231923ddb361b0daac293ea3cf/Focal_Loss.py#L42-L43

**Comment by yatengLG** [comment]
好的，谢谢。
